### PR TITLE
[FIX] point_of_sale: use compute_all properly for taxes computation

### PR DESCRIPTION
**Pull request by oco-odoo**
- Use the tax tags returned by the compute_all for the base lines instead of recomputing them separately

Splitting the computation caused inconsistent computation in some cases between the base and tax lines (one could select the refund distribution, while the other selected the invoice one).

- Pass the is_refund parameter to True if either the quantity or the price is negative (but not both)

Those negative lines should correspond to refund repartition (double-checked with the Belgian tax report in v12).

OPW 2351412

### CODE REVIEW

**Comment by pimodoo** [comment]
@caburj could you have a look at this fix?

**Comment by caburj** [comment]
@pimodoo Yes, I will.

**Comment by caburj** [comment]
I agree with the change on using the base_tags from compute_all. However, the part about is_refund is missing.

**Comment by oco-odoo** [comment]
@caburj shit, I screwed up my rebase >< Let me fix that ...

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -643,18 +644,21 @@ def get_income_account(order_line):
         # The 'is_refund' parameter is used to compute the tax tags. Ultimately, the tags are part
         # of the key used for summing taxes. Since the POS UI doesn't support the tags, inconsistencies
         # may arise in 'Round Globally'.
+        check_refund = lambda x: (x.qty < 0 or x.price_unit < 0) and not (x.qty < 0 and x.price_unit < 0)
```
Comment (caburj) [inline addons/point_of_sale/models/pos_session.py:647]: I think it is simplier if we multiple qty and price_unit and check if the product is less than 0, like so:
`check_refund = lambda line: line.qty * line.price_unit < 0`

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**